### DISCUSSION TITLE
Can we delete all tags from translated content yet?

### DISCUSSION THREAD

**Discussion by peterbe** [mdn/mdn-community#543] [Content]
This is a "fork" of https://github.com/mdn/content/discussions/5805 
That discussion talks about what to do with tags. We've learned the following there:

1. Only a handful of tags are actually used in en-us.
2. It seems the translated contents' tags are never used. E.g. the CSS sidebar for `fr` uses the documents in `en-us` to figure out which slugs to include. 
3. Tags are not used for site-search (peter's note: this one I'm 100% confident of)
4. We don't have any plans for doing site-search by tag (peter's note: this one I'm 99% confident of)
5. If we remove the `tags` key from the translated content front-matter, when built in KS, it will automatically pick up the en-us's `tags` from its front-matter. 
6. There are 5,263 translated `index.html` files that specify `tags:` in their front-matter. (it totals ~400KB as a combined string)

**I propose that we delete all `tags:` front-matter from all the translated content.** 
It will make it less confusing for translators who might think they have to keep this pristine and/or translated.

**Comment by peterbe** [comment]
Fishing for feedback
@wbamberg @Rumyra @SphinxKnight @Elchi3 @fiji-flo @escattone

**Reply by teoli2003** [reply]
Point 2 was done on purpose: that way untranslated new pages were appearing in red in sidebars and clicking on them lead to a call for translation.

This propose sounds like a good idea. (By the way, shouldn't `browser-compat-data` being maintained only on en-US too?)

**Reply by escattone** [reply]
I'm all for this, as long as the content folks agree.

**Reply by hamishwillee** [reply]
Vote +1 from me. 
- In fact you should remove **all** front matter that can be inferred from the US version. That would probably just leave the title and maybe the slug. Right? 
  [**Edit**: and I guess @teoli2003 agrees based on "By the way, shouldn't browser-compat-data being maintained only on en-US too?)"]
- Can we prevent tags from being re-added?

If we're not using them as tags for search or categorising all they can do is harm data quality. As discussed in mdn/mdn-community#524 it would be much better if we added separate meaningful keys for most of these.

**Reply by SphinxKnight** [reply]
>     * In fact you should remove **all** front matter that can be inferred from the US version. That would probably just leave the title and maybe the slug. Right?
>       [**Edit**: and I guess @teoli2003 agrees based on "By the way, shouldn't browser-compat-data being maintained only on en-US too?)"]

+1 for this. I'd be inclined to treat any inferred front matter material the same way and if confusion is a source of concern, as for myself, I'd be more confused having only the tags removed. Currently, adding a new localized docs consists of copying/pasting the en-US folder and adding the `translation_of` key,. Only dealing with tags could seem a bit strange.

I don't want to sidetrack here but **as I understand it, ultimately only `title` should differ from `en-US` to `localeX`.**

If we look at the front matter keys (that I know of), I'm not even sure `original_slug` should be kept theoretically speaking. For non-conflicting/orphaned docs: 

* `slug` : must be the same as `en-US` (can be inferred from the path in the tree)
* `tags` : needless to say more, can be deleted
* `translation_of` : key duplicates `slug` (again, for non-conflicting/orphaned docs)
* `original_slug` : lives better off in `redirects.txt`

> 
>     * Can we prevent tags from being re-added?
> 

This can be part from the review (though this does not prevent to add a check/flaw to help)

**Reply by peterbe** [reply]
That's right, only `title` makes sense. Everything else can and should be inferred. But how we get there is a different story. 

- `tags` is only ever used in kumascript rendering, but `slug` and `translation_of` etc. might be needed by the Yari builder before it even gets into the kumascript rendering. 
- the `translation_of` and `original_slug` might need to survive a little bit longer until we've `git rm` 'ed all the `orphaned` and `conflicting` translations. 
- If you rename a file in `en-us`, the existing translations won't necessarily know about that yet, but that's what `sync-translated-content` does even though I don't fully understand it. Or rather, I've already forgotten how it works. 
- For all content, en-US and the translated content, we want to get rid of the `slug` or at least only keep it if can't be exactly like the folder it sits in.

**Reply by hamishwillee** [reply]
Cool. I think you can delete the flags. I would find it useful to know that fixing the rest is still on your radar (e.g. an issue or whatever).

**Reply by peterbe** [reply]
As an update, I'm in the processing of deleting all the documents that have `conflicting` or `orphaned` in the slug. 
See https://github.com/mdn/translated-content/pulls?q=is%3Apr+%22delete+conflicting%22+
Once that's done, it definitely no longer makes sense for any of the translated content to have any front-matter that is *not* `title`. 
But I'd need to sleuth with @fiji-flo about what it means if you rename a en-US doc (which involves both `git mv ...` and editing the `slug`) but haven't done so yet with the translated content. 

Perhaps, with that in mind, it does *not* makes sense to start with deleting all the `tags` just to get that out of our hair because you'd be roping in a lot of translator reviewers twice then. 

In other news, I'm getting much closer now to enabling Preview URLs and the Review Companion to all translated content PRs. That'd make any cleanup efforts a lot smoother.

**Comment by pepelsbey** [comment]
Closing as outdated/irrelevant.
